users/userRoutes.js

```diff
@@ -0,0 +1,96 @@
+
+const express = require('express');
+const router = express.Router();
+const db = require('./userDb');
+
+function allCap(req, res, next){
+    req.body.name = req.body.name.toUpperCase();
+    next();
+}
+
+
+router.get('/', async (req, res)=>{
+    
+    const users = await db.get();
+    try{
+        res.status(200).json(users)
+    }
+    catch(er){
+        res.status(500).json({message: 'There was an error retrieving the data'})
+    }
+
+});
+
+router.get('/:id', async (req, res)=>{
+
+ let {id} = req.params;
+ let user = await db.get(id);
+
+ try{
+    res.status(200).json(user)
+ }
+ catch(er){
+    res.status(500).json({message: 'There was an error retrieving the data'})
+ }
+})
+
+router.get('/:id/posts', async (req, res)=>{
+    let {id} = req.params;
+    let userPosts = await db.getUserPosts(id);
+
+    try{
+        res.status(200).json(userPosts);
+    }
+    catch(er){
+        res.status(500).json({message: 'There was an error retrieving the data'})
+    }
+});
+
+
+router.post('/', allCap, (req, res)=>{
+    
+    let user = req.body;
+    
+    db.insert(user)
+    .then(r => res.status(200).json(r))
+    .catch(err => res.status(500).json({message: 'There was an error processing the data'}));
+
+})
+
+router.put('/:id', allCap, async (req, res)=>{
+    let {id} = req.params;
+    let post = req.body; 
+    let prevName = await db.get(id);
+    
+    db.update(id, user)
+    .then(r => res.status(200).json({message: `${prevName.name} has been changed to ${user.name}`}))
```

Comment:
I would recommend returning the updated resource, not a message like this. So the actual updated user object. Think about what is useful for a client side app.. generally a message made up of a string is super unuseful for a computer program..

also, you need to implement an error message if they try to edit a non-existent user